java-checks/src/test/files/checks/security/CryptographicKeySizeCheck.java

```diff
@@ -0,0 +1,124 @@
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import javax.crypto.KeyGenerator;
+
+class A {
+
+  public KeyGenerator keyG;
+  public KeyPairGenerator keyPairG = KeyPairGenerator.getInstance("RSA");;
+  private static final KeyGenerator keyG2 = KeyGenerator.getInstance("Blowfish");
+
+  public void foo1() throws NoSuchAlgorithmException {
+    KeyGenerator keyGen = KeyGenerator.getInstance("Blowfish");
+    keyGen.init(64);  // Noncompliant [[sc=5;ec=20]] {{Use a key length of at least 128 bits.}}
+  }
+
+  public void foo2() throws NoSuchAlgorithmException {
+    KeyGenerator keyGen = KeyGenerator.getInstance("Blowfish");
+    int x = 64;
+    keyGen.init(x); // FN
```

Comment:
for the false negatives, you can add a comment with the number of the ticket PY is working on (SONARJAVA-2491) and mention it can solve the problem (if my understanding is correct)

Reply:
Good point. Will make it explicit when solved that this is expected. And if no issue is raised when fixing the ticket, may raise some questions as well.

Reply:
Actually, SONARJAVA-2491 will enable us to read only the values of constant variables.
So, it will not solve the problem and I suspect that we need Symbolic Execution to fix the FNs of this rule.

Reply:
Cool, so it's an excellent reason to add a comment to be very explicit on why we are not detecting it, even if the case is trivial. For instance "requires following variables values with SE-based engine".